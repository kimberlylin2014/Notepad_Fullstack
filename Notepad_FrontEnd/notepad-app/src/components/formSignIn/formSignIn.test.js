import React from 'react';
import FormSignIn from './formSignIn.component';
import { shallow } from 'enzyme';

describe("Form Sign In Tests", () => {
    const wrapper = shallow(<FormSignIn />)
    it('Form Sign In SnapShot Test', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    it('Password Input should change value when OnChange is called', ()=> {
        const event = {
            target: {value: "hello", name: 'password'}
        }
        wrapper.find('[name="password"]').simulate("change", event)
        expect(wrapper.state('password')).toBe('hello')
    })

    it('Email Input should change value when OnChange is called', ()=> {
        const event = {
            target: {value: "hello", name: 'email'}
        }
        wrapper.find('[name="email"]').simulate("change", event)
        expect(wrapper.state('email')).toBe('hello')
    })

})