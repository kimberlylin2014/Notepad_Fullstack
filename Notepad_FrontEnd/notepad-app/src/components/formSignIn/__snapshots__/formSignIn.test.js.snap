// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Sign In Tests Form Sign In SnapShot Test 1`] = `
<div
  className="FormSignIn"
>
  <h3
    className="mb-4"
  >
    Sign In
  </h3>
  <Form
    tag="form"
  >
    <InputGroup
      tag="div"
    >
      <Input
        name="email"
        onChange={[Function]}
        placeholder="new username"
        type="text"
      />
      <InputGroupAddon
        addonType="append"
        tag="div"
      >
        <InputGroupText
          tag="span"
        >
          @notepad.com
        </InputGroupText>
      </InputGroupAddon>
    </InputGroup>
    <FormInput
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
    <Button
      className="mr-2"
      color="secondary"
      id="signIn-click"
      onClick={[Function]}
      tag="button"
    >
      Sign In
    </Button>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Register
    </Button>
  </Form>
</div>
`;
